### DISCUSSION TITLE
Copilot doesnt work when using Cloudflare WARP client

### DISCUSSION THREAD

**Discussion by jay-steyn** [community/community#52052] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

We have an issue where our WARP client is blocking access to Copilot.
Is there a list anywhere of endpoints or IP addresses used by Copilot that we can use to whitelist?

**Comment by Akash1134** [comment, marked as answer]
Hi @JaySteyn ,

You can finetune your network traffic using the following endpoints in firewall rules:

- https://github.com/login/device/code
- https://github.com/login/oauth/access_token
- https://api.github.com/user
- https://api.github.com/copilot_internal/v2/token
- https://api.github.com/copilot_internal/notification
- https://default.exp-tas.com
- https://copilot-telemetry.githubusercontent.com/telemetry
- https://copilot-proxy.githubusercontent.com

Hope that helps! Thanks!

**Reply by jay-steyn** [reply]
thanks! will give that a try.
